airflow/models/dagrun.py

```diff
@@ -252,19 +252,22 @@ def refresh_from_db(self, session: Session = NEW_SESSION) -> None:
 
     @classmethod
     @provide_session
-    def active_runs_of_dags(cls, dag_ids=None, only_running=False, session=None) -> Dict[str, int]:
+    def active_runs_of_dags(
+        cls,
+        dag_ids: Optional[Iterable[str]] = None,
+        only_running: bool = False,
+        session: Session = NEW_SESSION,
+    ) -> Dict[str, int]:
         """Get the number of active dag runs for each dag."""
         query = session.query(cls.dag_id, func.count('*'))
         if dag_ids is not None:
-            # 'set' called to avoid duplicate dag_ids, but converted back to 'list'
-            # because SQLAlchemy doesn't accept a set here.
-            query = query.filter(cls.dag_id.in_(list(set(dag_ids))))
+            query = query.filter(cls.dag_id.in_(dag_ids))
         if only_running:
-            query = query.filter(cls.state == State.RUNNING)
+            query = query.filter(cls.state == DagRunState.RUNNING)
         else:
-            query = query.filter(cls.state.in_([State.RUNNING, State.QUEUED]))
+            query = query.filter(cls.state.in_([DagRunState.RUNNING, DagRunState.QUEUED]))
         query = query.group_by(cls.dag_id)
-        return {dag_id: count for dag_id, count in query.all()}
```

Comment:
@uranusjr i think this was probably a mistake ... i.e. the removal of `.all()`, no?

Reply:
i reverted the changes to this file for now.  i don't think they are required by your other changes.